paddle/fluid/ir/dialect/pd_attribute.cc

```diff
@@ -16,9 +16,13 @@
 
 namespace paddle {
 namespace dialect {
-phi::IntArray IntArrayAttribute::data() const { return storage()->GetAsKey(); }
+const phi::IntArray& IntArrayAttribute::data() const {
+  return storage()->GetAsKey();
+}
 
-phi::DataType DataTypeAttribute::data() const { return storage()->GetAsKey(); }
+const phi::DataType DataTypeAttribute::data() const { 
```

Comment:
```suggestion
const phi::DataType& DataTypeAttribute::data() const { 
```

同时storage 里对应的 GetAsKey 函数也要返回引用才行，否则返回一个临时对象的上引用是危险的。